tinygrad/ops.py

```diff
@@ -263,20 +263,40 @@ def exec_ast(self, ast:LazyOp, output, inputs, var_vals, **kwargs):
     # compilation time
     def get_program():
       from tinygrad.codegen.linearizer import Linearizer
+      from tinygrad.features.search import beam_search, time_linearizer
+
       k = Linearizer(ast, self.linearizer_opts)
       assert k.info.dtype == output.dtype, f"linearizer must match dtype. linearizer wants {k.info.dtype} but buffer is {output.dtype}"
       if not getenv("NOOPT"):
-        if not k.apply_tensor_cores(getenv("TC", 1)): k.hand_coded_optimizations()
+        ktc = k.copy()
```

Comment:
I think this should only happen if we are searching with BEAM, otherwise listen to TC